devices/StepperMotor/Threading_StepperMotor.py

```diff
@@ -0,0 +1,522 @@
+from PyQt5 import QtWidgets, QtCore, QtGui
+import sys, os
+from time import sleep
+import RPi.GPIO as GPIO
+
+from PyQt5.QtGui import *
+from PyQt5.QtWidgets import *
+from PyQt5.QtCore import *
+
+# TO DO:  the current position should update for every step  
+
+cwd = os.getcwd()
+print(cwd)
+# Check if '15tw-smartsystem' is in the components
+if "15tw-smartsystem" not in cwd.split(os.path.sep):
+    raise ValueError("The directory does not contain '15tw-smartsystem' folder.")
+
+# Rebuild the directory string up to and including '15tw-smartsystem', prevent import errors
+cwd = os.path.sep.join(
+    cwd.split(os.path.sep)[: cwd.split(os.path.sep).index("15tw-smartsystem") + 1]
+)
+sys.path.insert(0, cwd)
+
+
+from devices.StepperMotor.Hakeem_StepperMotors import (Ui_Form) 
+
+# Define GPIO pins and constants
+global DIR, STEP, CW, CCW, log
```

Comment:
Also we shouldn't use global variables, three letter variables can quickly get overwritten when this goes into a larger diagnostic system and that will be a nightmare to debug